Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm

```diff
@@ -2464,6 +2476,44 @@ - (id)accessibilityHitTest:(NSPoint)point
         hit = self;
 
     return NSAccessibilityUnignoredAncestor(hit);
+
+}
+
+- (void)accessibilityHitTestResolvingRemoteFrame:(NSPoint)point callback:(void(^)(NSString *))callback
+{
+    if (!AXObjectCache::clientIsInTestMode()) {
+        callback(@"");
+        return;
+    }
+
+    id hitTestResult = [self accessibilityHitTest:point];
+    if (!hitTestResult) {
+        callback(@"");
+        return;
+    }
+
+    if ([hitTestResult isKindOfClass:[NSAccessibilityRemoteUIElement class]]) {
+        RefPtr<AXCoreObject> backingObject = self.updateObjectBackingStore;
+        if (!backingObject)
+            return;
+
+        auto* axObject = backingObject->accessibilityHitTest(IntPoint(point));
+        if (axObject && axObject->isRemoteFrame()) {
+            RefPtr page = backingObject ? backingObject->page() : nullptr;
+            AXRemoteFrame* axRemoteFrame = dynamicDowncast<AXRemoteFrame>(axObject);
+
+            auto clientCallback = [callback = makeBlockPtr(callback)] (String result) {
+                callback(nsStringNilIfEmpty(result));
+            };
```

Comment:
Doing this is unnecessary if `!page` or `!axRemoteFrame` — can we move it inside that scope?

```cpp
if (page && axRemoteFrame) {
    auto clientCallback = [callback = makeBlockPtr(callback)] (String result) {
        callback(nsStringNilIfEmpty(result));
    };
    page->chrome().client().resolveAccessibilityHitTestForTesting(*axRemoteFrame->frameID(), IntPoint(point), WTFMove(clientCallback));
}
```